projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -752,15 +752,22 @@ function jetpack_filter_excerpt_for_newsletter( $excerpt, $post = null ) {
  *
  * @return string
  */
-function maybe_get_locked_content( $the_content ) {
+function add_paywall( $the_content ) {
+	$block_name = 'jetpack/paywall';
 	require_once JETPACK__PLUGIN_DIR . 'modules/memberships/class-jetpack-memberships.php';
 
 	if ( Jetpack_Memberships::user_can_view_post() ) {
 		return $the_content;
 	}
 
 	$post_access_level = Jetpack_Memberships::get_post_access_level();
-	return get_locked_content_placeholder_text( $post_access_level );
+	$paywalled_content = get_paywall_blocks( $post_access_level );
+
+	if ( has_block( $block_name ) ) {
```

Comment:
Checked that `has_block` has been available since WP 5.0 so we're good not to check if it exists. :-) 

https://github.com/WordPress/WordPress/blob/dcc82c39ccfc4d0aef8751b317724fea0c27b40f/wp-includes/blocks.php#L632C11-L632C16